app.json

```diff
@@ -0,0 +1,32 @@
+{
+  "id": "b944c053-d2b5-4acf-ad80-8d27cea5e378",
+  "name": "BCRealworldPractice",
+  "publisher": "Default Publisher",
+  "version": "1.0.0.0",
+  "brief": "",
+  "description": "",
+  "privacyStatement": "",
+  "EULA": "",
+  "help": "",
+  "url": "",
+  "logo": "",
+  "dependencies": [],
+  "screenshots": [],
+  "platform": "1.0.0.0",
+  "application": "20.0.0.0",
+  "idRanges": [
+    {
+      "from": 50400,
+      "to": 50499
+    }
+  ],
+  "resourceExposurePolicy": {
+    "allowDebugging": true,
+    "allowDownloadingSource": false,
+    "includeSourceInSymbolFile": true
+  },
+  "runtime": "9.0",
```

Comment:
what does runtime mean? and what is the importance? why do we need 13.0 for BC24?

Reply:
runtime is a property which define what version of AL Language is applied to project
Different versions of BC have different version AL language.

For example BC20 has runtime 9.0 and AL language version 9.0, so you cant use some methods from newest versions of AL language, for example Record.ReadIsolation method will be unavailable for the project because it appears in 12.0 (if I remember correctly)

BC24 use AL version 13.0+

You can focus any property in app.json to have some tooltips

![image](https://github.com/Ams272/BC-Practice/assets/16802407/a0203865-6b68-45b0-80b6-318d79a53abd)

Also, if you clear value in the value of app.json you can use ctrl+space to get suggested options, each option have tooltip, you can move to next option via arrows ok keyboard
ctrl+space combination is very important to get list of available options in AL language

![image](https://github.com/Ams272/BC-Practice/assets/16802407/b3ca68e8-fa39-47ae-a3c1-177e77d7a2f6)

Reply:
Ohhh i very much understand the principle now.